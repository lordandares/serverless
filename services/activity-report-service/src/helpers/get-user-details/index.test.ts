const mockToArray = jest.fn()
const mockSort = jest.fn().mockReturnValue({ toArray: mockToArray })
const mockFind = jest.fn().mockReturnValue({ sort: mockSort })
const mockFindOne = jest.fn()

const mockGetCollection = jest.fn().mockResolvedValue({
  find: mockFind,
  findOne: mockFindOne,
})

import { mongo } from '@lighthouse/serverless-common'

import { getUserDetails } from './'

mongo.getCollection = mockGetCollection

describe('helpers:getUserDetails', () => {
  const MOCK_APPLICATION = {
    _id: '565fe59257c5d59e60cc35b1',
    name: 'Testing Application',
  }

  const MOCK_USER = {
    application: '565fe59257c5d59e60cc35b5',
    email: 'testing@test.com',
    firstName: 'Testing',
    preferences: {
      notifications: {
        channels: [
          {
            enabled: true,
            name: 'daily-location-report',
            options: {
              hours: 9,
              timezone: 'America/Los_Angeles',
            },
          },
        ],
      },
    },
    permissions: [
      {
        module: 'location',
        type: 'document',
        value: '565fe59257c5d59e60cc35b3',
      },
      {
        module: 'zone',
        type: 'module',
        value: '565fe59257c5d59e60cc35b6',
      },
    ],
  }

  afterEach(() => jest.clearAllMocks())

  it('it returns user details', async () => {
    expect.assertions(4)

    mockFindOne.mockResolvedValue(MOCK_APPLICATION)

    const result = await getUserDetails(MOCK_USER)

    expect(mockGetCollection).toHaveBeenCalledTimes(1)
    expect(mockGetCollection).toBeCalledWith('applications')

    // application query
    expect(mockFindOne).toBeCalledWith({
      _id: new mongo.ObjectId(MOCK_USER.application),
    })

    expect(result).toMatchSnapshot()
  })

  it('handles error when returning applications and returns false', async () => {
    expect.assertions(1)

    const error = new Error('Application error')
    mockFindOne.mockRejectedValue(error)

    const result = await getUserDetails({
      ...MOCK_USER,
      application: '565fe59257c5d59e60cc35d1',
    })

    expect(result).toEqual(undefined)
  })
})
