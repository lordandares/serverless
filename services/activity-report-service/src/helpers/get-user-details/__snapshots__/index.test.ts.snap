// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers:getUserDetails it returns user details 1`] = `
Object {
  "applicationName": "Testing Application",
  "email": "testing@test.com",
  "firstName": "Testing",
  "lastName": undefined,
  "preferences": Object {
    "notifications": Object {
      "channels": Array [
        Object {
          "enabled": true,
          "name": "daily-location-report",
          "options": Object {
            "hours": 9,
            "timezone": "America/Los_Angeles",
          },
        },
      ],
    },
  },
  "timezone": "America/Los_Angeles",
}
`;
